rero_ils/modules/documents/views.py

```diff
@@ -342,17 +327,26 @@ def item_library_pickup_locations(item):
     """Get the pickup locations of the library of the given item."""
     location_pid = item.replace_refs()['location']['pid']
     location = Location.get_record_by_pid(location_pid)
-    library_pid = location.replace_refs()['library']['pid']
-    library = Library.get_record_by_pid(library_pid).replace_refs()
-    organisation = Organisation.get_record_by_pid(
-        library['organisation']['pid'])
-    locations = []
-    for library in organisation.get_libraries():
-        location = Location.get_record_by_pid(
-            library.get_pickup_location_pid())
-        if location:
-            locations.append(location)
-    return locations
+    # Either the location defines some 'restrict_pickup_to' either not.
+    # * If 'restrict_pickup_to' is defined, then only these locations are
+    #   eligible as possible pickup_locations
+    # * Otherwise, get all organisation pickup locations of the item belongs to
+    if 'restrict_pickup_to' in location:
+        # Get all pickup locations as Location objects and append it to the
+        # location item (removing possible None values)
+        pickup_locations = list(filter(None, [
+            Location.get_record_by_pid(loc_pid)
+            for loc_pid in location.restrict_pickup_to
+        ]))
+        return pickup_locations
+    else:
+        org = Organisation.get_record_by_pid(location.organisation_pid)
+        # Get the pickup location from each library of the item organisation
+        # (removing possible None value)
+        return list(filter(None, [
```

Comment:
The first argument of filter should be a function: probably you want something like:
```python
filter(lambda x: x is not None, my_list)
```

Reply:
It's a Python shortcut to filter list removing `None` elements : 
```
>>> my_list = [None, 1, 'test', 'None', None]
>>> my_list = list(filter(None, my_list))
>>> print(my_list)
[1, 'test', 'None']
```
It is for me the simplest way here to filter possible None location (should never happens)